### ISSUE TITLE
[Bug] ConfigurationDataUtils does not throw exception on unknown properties anymore

### ISSUE THREAD

**Issue by alpreu** [apache/pulsar#23792]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.

### Version

Affected LTS versions are:
pulsar-client-all:3.0.7
pulsar-client-all:4.0.1

### Minimal reproduce step

GitHub repository for reproducing the issue: TODO

or alternatively,

Create a configuration `HashMap` containing unknown key-value pairs, then try to load it like this:
```
ClientConfigurationData conf =  ConfigurationDataUtils.loadData(
                configuration, new ClientConfigurationData(), ClientConfigurationData.class);
```

### What did you expect to see?

RuntimeException with "Failed to load config into existing configuration data" to be thrown

### What did you see instead?

No exception is thrown, the `ClientConfigurationData` object is instantiated

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by pdolif** [comment]
I think it is related to #22541 / https://github.com/apache/pulsar/pull/22541/commits/c0c01c2677d9f68933f20666bbc66bee443de89c. 
There, `@JsonIgnoreProperties(ignoreUnknown = true)` was added to `ClientConfigurationData`.

**Comment by alpreu** [comment]
> I think it is related to #22541 / [c0c01c2](https://github.com/apache/pulsar/commit/c0c01c2677d9f68933f20666bbc66bee443de89c). There, `@JsonIgnoreProperties(ignoreUnknown = true)` was added to `ClientConfigurationData`.

Yes, looks like it. @lhotari was this an intentional change?

**Comment by lhotari** [comment]
> > I think it is related to #22541 / [c0c01c2](https://github.com/apache/pulsar/commit/c0c01c2677d9f68933f20666bbc66bee443de89c). There, `@JsonIgnoreProperties(ignoreUnknown = true)` was added to `ClientConfigurationData`.
> 
> Yes, looks like it. @lhotari was this an intentional change?

@alpreu Yes, it was intentional at the time. I'll have to dig the reason. 
What's the actual problem that it's causing? Please share more context about your use case. What makes your use case dependent on throwing an exception?

**Comment by alpreu** [comment]
> > > I think it is related to #22541 / [c0c01c2](https://github.com/apache/pulsar/commit/c0c01c2677d9f68933f20666bbc66bee443de89c). There, `@JsonIgnoreProperties(ignoreUnknown = true)` was added to `ClientConfigurationData`.
> > 
> > 
> > Yes, looks like it. @lhotari was this an intentional change?
> 
> @alpreu Yes, it was intentional at the time. I'll have to dig the reason. What's the actual problem that it's causing? Please share more context about your use case. What makes your use case dependent on throwing an exception?

It's not a big problem in that sense but having the exception in place as nice for us because we use plain `Map`s to configure connections, so when a user misspelled a parameter we could easily detect that, while now it would silently be discarded and they might wonder why their connection does not work
